### PR TITLE
Add load_value() and dump_value() for value serialization

### PR DESCRIPTION
**Pull request by Pliner**
## Summary

Adds `load_value()` and `dump_value()` functions for serializing standalone values (primitives and collections) without requiring dataclass wrappers.

## Features

- `mr.load_value(type[T], data)` - deserialize any value type
- `mr.dump_value(type[T], data)` - serialize any value type
- Support for primitives (`int`, `str`, `bool`, `float`)
- Support for optional primitives (`int | None`, `str | None`, etc.)
- Support for collections (`list`, `dict`, `set`, `frozenset`, `tuple`)
- Support for complex types (`decimal.Decimal`, `uuid.UUID`, `datetime.datetime`)
- Nested dataclasses inside collections work correctly (`list[MyDataclass]`)
- **Raises `ValueError`** if root type is a dataclass (use `dump()`/`load()` for those)

## Implementation

Uses a generic wrapper dataclass (`_Wrapper[T]`) internally to reuse existing schema generation machinery.

## Test plan

- [x] Parametrized tests covering primitives, collections, optional types, complex types
- [x] Edge case tests for empty collections
- [x] Tests for `ValueError` on dataclass root type
- [x] All existing tests pass
- [x] `make lint` passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: examples/13_value_serialization.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+# Lists
+dumped = mr.dump_value(list[int], [1, 2, 3])  # Returns: [1, 2, 3]
+loaded = mr.load_value(list[int], [1, 2, 3])  # Returns: [1, 2, 3]
+
+# Dictionaries
+dumped = mr.dump_value(dict[str, int], {"a": 1, "b": 2})  # Returns: {"a": 1, "b": 2}
+loaded = mr.load_value(dict[str, int], {"a": 1, "b": 2})  # Returns: {"a": 1, "b": 2}
+
+# Sets (serialized as lists)
+dumped = mr.dump_value(set[int], {1, 2, 3})  # Returns: [1, 2, 3]
```
Comment (vladimir-kozyrev) [inline examples/13_value_serialization.md]: Shouldn't it return a set?
```suggestion
dumped = mr.dump_value(set[int], {1, 2, 3})  # Returns: {1, 2, 3}
```

Comment (vladimir-kozyrev) [reply examples/13_value_serialization.md]: Ah, never mind. I get it now.
